src/v2/Apps/Conversation/Components/Reply.tsx

```diff
@@ -50,51 +50,81 @@ interface ReplyProps {
 export const Reply: React.FC<ReplyProps> = props => {
   const { environment, conversation } = props
   const [buttonDisabled, setButtonDisabled] = useState(true)
+  const [loading, setLoading] = useState(false)
+  const [showModal, setShowModal] = useState(false)
   const textArea = useRef()
 
+  const setupAndSendMessage = () => {
+    {
+      setLoading(true)
+      return SendConversationMessage(
+        environment,
+        conversation,
+        // @ts-ignore
+        textArea?.current?.value,
+        _response => {
+          // @ts-ignore
+          textArea.current.value = ""
+          setLoading(false)
+          setButtonDisabled(true)
+        },
+        _error => {
+          setLoading(false)
+          setShowModal(true)
+        }
+      )
+    }
+  }
+
   return (
-    <StyledFlex p={1} right={[0, null]} zIndex={2}>
-      <FullWidthFlex width="100%">
-        <StyledTextArea
-          onInput={event => {
-            const field = event.target as HTMLTextAreaElement
-            field.style.height = "inherit"
-            if (buttonDisabled && field.value.length > 2) {
-              setButtonDisabled(false)
-            }
-            if (!buttonDisabled && field.value.length <= 2) {
-              setButtonDisabled(true)
-            }
-            const height = field.scrollHeight
-            field.style.height = height + "px"
-          }}
-          placeholder="Type your message"
-          ref={textArea}
-        />
-      </FullWidthFlex>
-      <Flex alignItems="flex-end">
-        <StyledButton
-          disabled={buttonDisabled}
-          onClick={_event => {
-            return SendConversationMessage(
-              environment,
-              conversation,
-              // @ts-ignore
-              textArea?.current?.value,
-              _response => {
-                // @ts-ignore
-                textArea.current.value = ""
+    <>
+      <Dialog
```

Comment:
This looks like a big diff but adding `ignore whitespace` flag it just adds `<Dialog />` component.